### PR TITLE
ADT, diagnostic plots update

### PR DESCRIPTION
**Pull request by erflynn**
added params for ADT background filtering, updated diagnostic plots to use log-scale and be smaller

### CODE REVIEW

**Comment by erflynn** [comment]
@syellamilli and @dtm2451  - added the ADT features finally, also fixed the diagnostic plots. not sure if we should merge this in yet? pro -- proper ADT filtering, better (log-scale)/smaller plots; con -- will change output

also fyi tests are not working at this pt, so not bothering to update snapshot until things are fixed

**Comment by erflynn** [comment]
added @tastam's doublet plot. 
I also added "final" post-filter plots -- since the "post" plots are just the original pre plots with the selected filters. Should we instead of adding those post plots, overwrite the pre? Or do something else? I'm not sure.

**Comment by erflynn** [comment]
Currently the plots that we have are:
- "pre": all cells (w default cuts) - log scale for counts/features
- "post": all cells (w selected cuts) - log scale for counts/features
- "doublet": filtered cells before DBL removal - log + non-log scale, nCount_ADT/RNA
- "final": filtered cells after DBL removal - log scale for counts/features
Does this make sense? should we shift this around? (Before it was just pre/post)
Do we want non-log scale plots?

**Comment by erflynn** [comment]
[] TODO: isotype-ctl diagnostic plot?

**Comment by dtm2451** [comment]
First...
I may be wrong here cuz I've had to remake "post" plots for zooming in and probably just changed what my self-made ones showed without realizing, but 
> - "post": all cells (w selected cuts) - log scale for counts/features

sounds like a change to what the "post" plots used to be?  And the name is confusing then cuz it's not post-"filter" as I always assumed this "post" meant.

I wonder if we should change those names to "default" and "reviewed" or something...

**Overall though,** I like the set! but I do think non-log is helpful for knowing the non-log number when setting cutoffs.

Would it be overkill to append "log" in the filename for log-scale versions and produce both log and non-log for all?

**Comment by erflynn** [comment]
good ideas. 
I also thought they were post filter, but when I was working on this this AM, I realized they are updated cutoffs but not post filter. :/

I think we can make log and non-log & change names to default and reviewed! thank you.

**Comment by erflynn** [comment]
here is an example doublet plot for CITE-seq in case helpful image

**Comment by erflynn** [comment]
I'd also like to some automated thresholding suggestions, but I dont know if it's worth it since we'd need a new R container (reticulate issue, works for v4 Rstudio but not v4 command line). I wonder if it would make things too busy too?

**Comment by erflynn** [comment]
Added isotype control and background plots, example here:
AIP1-POOL-DB15-SCG1_adt_diagnostic_plots

The key difference between the background nCount_RNA/nCount_ADT and the original plot is that this does not filter by min.cells and min.genes, so we can actually see the full background distribution.

**Comment by erflynn** [comment]
Some thoughts on the ADT plots (we can discuss at a mtg, writing so as not to forget):
- do we like the text? should it be different?
- do we want non-log-scale as well?
- should we move the other ADT plots from the standard diagnostic plots into here?
- should we have a reviewed and filtered version of these

**Comment by erflynn** [comment]
Example full diagnostic plots
image

File: single_cell_RNAseq/bin/filter_barcodes_pre_fmx.R [hunk 1]
```diff
@@ -52,11 +52,18 @@ if (adt.present){
...
+num_rows = ifelse(adt.present, 5, 3)
+merge = ggarrange(plotlist=plot_list, ncol=4,nrow=num_rows)
+ggsave(merge, file=sprintf("%s_diagnostic_plots_reviewed.png", LIBRARY) , width=30, height=7*num_rows, bg="white", dpi=72)
+
+if ("cellranger_cell" %in% colnames(sobj@meta.data)){
+  plot_list = suppressWarnings(make_plots(sobj, params, adt.present, group="cellranger_cell"))
+  num_rows = ifelse(adt.present, 5, 3)
+  merge = ggarrange(plotlist=plot_list, ncol=4,nrow=num_rows)
+  ggsave(merge, file=sprintf("%s_cr_diagnostic_plots_reviewed.png", LIBRARY) , width=30, height=7*num_rows, bg="white", dpi=72)
+}
```
Comment (dtm2451) [inline single_cell_RNAseq/bin/filter_barcodes_pre_fmx.R:65]: Super tiny nit here, so feel free to ignore if the work you do to test makes it feel not worth it to just save a few lines of code...

Seems like you could start this chunk with 
```
if ("cellranger_cell" %in% colnames(sobj@meta.data)) {
    plot_list = suppressWarnings(make_plots(sobj, params, adt.present, group="cellranger_cell"))
} else {
    plot_list = suppressWarnings(make_plots(sobj, params, adt.present))
}
```
and then remove the duplication of the num_rows, merge, and ggsave lines which all look the same.

**Comment by dtm2451** [comment]
Heyo!  I've read through your new comments here and looked through some changes, but don't have time for a full code review, or to wrap my head around the plot output differences, today.

Minor recs I do have:
- I forget if there is a doc somewhere that describes what the values in cutoffs.yml represent, but I think it would be important to explain the behavior of the new `background_*` values to users somewhere.  I don't see that in the diffs.
- (Super minor nit) Make Al, Shiv, and I reviewers instead of assignees.  Doesn't really matter much, but I think of assignee as whoever is in charge of the code changes, a.k.a. you unless the intent was to say that you need to hand this off?

**Comment by erflynn** [comment]
Thanks so much for the feedback! I'm going to go ahead merge this PR.

**Pull request merged**